core/tests/protractor.conf.js

```diff
@@ -319,9 +323,52 @@ exports.config = {
     // will be available. For example, you can add a Jasmine reporter with:
     //     jasmine.getEnv().addReporter(new jasmine.JUnitXmlReporter(
     //         'outputdir/', true, true));
-    var _ADD_SCREENSHOT_REPORTER = true;
 
-    if (_ADD_SCREENSHOT_REPORTER) {
+    var spw = '';
+    var uniqueString = '';
+    var vidPath = '';
+    // Enable allVideos if you want success videos to be saved.
+    var allVideos = false;
+
+    // Only running video recorder on Github Actions.
+
+    if (process.env.PWD == '/home/runner/work/oppia/oppia') {
+      jasmine.getEnv().addReporter({
+        specStarted: function(){
+          let ffmpegArgs = [
+            '-y',
+            '-r', '30',
+            '-f', 'x11grab',
+            '-s', '1280x1024',
+            '-i', process.env.DISPLAY,
+            '-g', '300',
+            '-loglevel', '16',
+          ];
+          uniqueString = randomString.generate(7);
+          var name = uniqueString + '.mp4';
+          var dirPath = path.resolve('__dirname', '..', '..', 'protractor-video/');
+          try {
+            fs.mkdirSync(dirPath, { recursive: true });
+          } catch (err) {}
+          vidPath = path.resolve(dirPath, name);
+          console.log(`Video path: ${vidPath}`);
+          ffmpegArgs.push(vidPath);
+          spw = childProcess.spawn('ffmpeg', ffmpegArgs);
+          spw.on('close', function(code){console.log(`ffmpeg exited with code ${code}`)});
+        },
+        specDone: function(result) {
+          spw.kill();
+          if (result.status == 'passed' && !allVideos && fs.existsSync(vidPath)) {
+            fs.unlinkSync(vidPath);
+            console.log('Video was deleted successfully (test passed).');
+          }
+        },
+      });
+    }
+
+    // Screenshots will only run on CircleCI.
+
+    if (process.env.PWD == '/home/circleci/oppia') {
```

Comment:
Here, we can use the `CIRCLECI` variable ([docs](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables))

Reply:
done